target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-sxr80.dts

```diff
@@ -0,0 +1,643 @@
+	interrupt-parent = <&intc>;
+
+	aliases {
+		serial0 = &blsp1_uart5;
+		led-boot = &led_front_blue;
+		led-failsafe = &led_front_red;
+		led-running = &led_front_green;
+		led-upgrade = &led_front_white;
+		/* Aliases as required by u-boot to patch MAC addresses */
+		ethernet0 = "/soc/dp1";
+		ethernet1 = "/soc/dp2";
+		ethernet2 = "/soc/dp3";
+		ethernet3 = "/soc/dp4";
+		ethernet6 = "/soc/dp5";
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+		bootargs-append = " root=/dev/ubiblock0_1";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 54 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&tlmm 57 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_front_blue: front-blue {
+			label = "blue:front";
+			gpios = <&tlmm 33 GPIO_ACTIVE_LOW>;
+		};
+
+		led_front_green: front-green {
+			label = "green:front";
+			gpios = <&tlmm 29 GPIO_ACTIVE_LOW>;
+		};
+
+		led_front_red: front-red {
+			label = "red:front";
+			gpios = <&tlmm 31 GPIO_ACTIVE_LOW>;
+		};
+
+		led_front_white: front-white {
+			label = "white:front";
+			gpios = <&tlmm 26 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_green: power-green {
+			label = "green:power";
+			gpios = <&tlmm 21 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		led_power_red: power-red {
+			label = "red:power";
+			gpios = <&tlmm 22 GPIO_ACTIVE_LOW>;
+			panic-indicator;
+		};
+
+	};
+};
+
+&tlmm {
+	mdio_pins: mdio-pins {
+		mdc {
+			pins = "gpio68";
+			function = "mdc";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+
+		mdio {
+			pins = "gpio69";
+			function = "mdio";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+	};
+
+	i2c_0_pins: i2c-0-pinmux {
+		mux {
+			pins = "gpio42", "gpio43";
+			function = "blsp1_i2c";
+			drive-strength = <8>;
+			bias-disable;
+		};
+	};
+
+	leds_pins: leds_pinmux {
+		led_power_green {
+			pins = "gpio21";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_power_red {
+			pins = "gpio22";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_white {
+			pins = "gpio26";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_green {
+			pins = "gpio29";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_red {
+			pins = "gpio31";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_blue {
+			pins = "gpio33";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+	};
+
+
+};
+
+&blsp1_uart5 {
+	status = "okay";
+};
+
+&blsp1_i2c2 {
+	pinctrl-0 = <&i2c_0_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+	
+	tlc59208f@27 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "ti,tlc59108";
+		reg = <0x27>;
+		
+		led0@0 {
+			label = "orbi:led0:rgb";
+			reg = <0x0>;
+			linux,default_trigger = "default-on";
```

Comment:
Any idea why this isn't working? The LEDs just stay off and trigger is "none". According to https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/leds/leds-tlc591xx.txt this is the way to do it (and that txt hasn't been touched for 6 years, so it should be in the linux version we're using).

Reply:
Is this the same controller as in SRK60 Orbi?
No need for ledN@N naming, just led@N is fine, also the labels should be "color:name".
The reg property should be decimal, DTC will convert it to hex anyway.

The default trigger property is incorrect, it needs to be `linux,default-trigger` but for just defaulting to ON its not correct anyway.
You should use `default-state = "on"`

The TCL binding doc is completely outdated, just use the generic bindings instead:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/Documentation/devicetree/bindings/leds/common.yaml?h=v5.10.78

Reply:
Are you sure that `default-state = "on"` works for the TCL? I tested it and it didn't work. In the file you linked it says under the default-on trigger
```
# LED will turn on (but for leds-gpio see "default-state" property in
# Documentation/devicetree/bindings/leds/leds-gpio.yaml)
```
So maybe default-state is just for gpio-based LEDs? Also in 2cb24b3f3cd89692f3c0bd137f3f560ada359bfa the trigger was used, I believe it is the same as in the SRK60.

Reply:
They are both parsed by the LED core and should work, however if `default-state` doesn't then feel free to use the `linux,default-trigger`

Reply:
`linux,default-trigger` works. Fixed.